lib/injector.rb

```diff
@@ -0,0 +1,15 @@
+class Array
+
+  def injector(accum = 0)
+    return nil if self.empty?
+    if block_given?
+      self.each {|element| accum = yield(accum, element)}; accum
```

Comment:
Try not to use semi-colons instead of splitting stuff onto separate lines - the fact that your method is that huge is probably a code smell